File: app/src/main/java/com/example/gymlog/ui/MainActivity.java
package com.example.gymlog.ui;


import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.gymlog.R;
import com.example.gymlog.database.AppDatabase;
import com.example.gymlog.database.ExerciseDAO;
import com.example.gymlog.database.ExerciseResultDao;
import com.example.gymlog.database.TestKt;
import com.example.gymlog.ui.exercises.activities.ExerciseManagementActivity;
import com.example.gymlog.ui.programs.FitnessProgramsActivity;

import java.util.concurrent.Executors;


public class MainActivity extends AppCompatActivity {

    private AppDatabase db;
    private ExerciseResultDao resultDao;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });



        findViewById(R.id.cardNewExercises).setOnClickListener(v -> {
            Log.d("LogTag", "New Exercises clicked");
            startActivity(new Intent(MainActivity.this, ExerciseManagementActivity.class));
        });

        findViewById(R.id.cardPrograms).setOnClickListener(v -> {
            Log.d("LogTag", "New Exercises clicked");
            startActivity(new Intent(MainActivity.this, FitnessProgramsActivity.class));
        });


        ExerciseDAO exerciseDAO = new ExerciseDAO(this);
        exerciseDAO.logAllExercises();





        // код тесту бд коли немає suspend в DAO
//        Executors.newSingleThreadExecutor().execute(() -> {
//            try{
//                TestKt.testDatabase(getApplicationContext());
//            }catch (Exception e){
//                Log.d(TestKt.TAG,"Помилка роботи з БД", e);
//            }
//        });



        TestKt.testDatabaseWithCallback(getApplicationContext(), () -> {
            Log.d(TestKt.TAG, "Операція завершена");
            return null;
        });



        TestKt.testDatabaseJavaWrapper(getApplicationContext(), () -> {
            Log.d(TestKt.TAG, "Операція завершена");
            return null;
        });




    }



}

File: app/src/main/java/com/example/gymlog/ui/MainActivity.kt
package com.example.gymlog.ui

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import com.example.gymlog.R
import com.example.gymlog.database.ExerciseDAO
import com.example.gymlog.database.testDatabaseJavaWrapper
import com.example.gymlog.ui.exercises.activities.ExerciseManagementActivity
import com.example.gymlog.ui.programs.FitnessProgramsActivity
import com.example.gymlog.ui.theme.MyAppTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        // Включаємо edge-to-edge режим
        WindowCompat.setDecorFitsSystemWindows(window, false)
        super.onCreate(savedInstanceState)

        // Ініціалізація бази даних
        val exerciseDAO = ExerciseDAO(this)
        exerciseDAO.logAllExercises()

        //тест бази Room
        testDatabaseJavaWrapper(
            this,
            applicationContext
        ) {
            runOnUiThread {
                Toast.makeText(this, "Операція завершена", Toast.LENGTH_SHORT).show()
            }
            null
        }

        setContent {
            // Використовуємо MyAppTheme для автоматичного підбору кольорів
            MyAppTheme {
                MainScreen()
            }
        }
    }
}

@Composable
fun MainScreen() {
    val context = LocalContext.current

    // Використовуємо кольори з MaterialTheme, які встановлюються через MyAppTheme
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .padding(WindowInsets.systemBars.asPaddingValues()),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.SpaceEvenly
    ) {
        // Заголовок
        Text(
            text = stringResource(id = R.string.my_gym_log),
            fontSize = 24.sp,
            color = MaterialTheme.colorScheme.onBackground,
            modifier = Modifier.padding(top = 16.dp)
        )

        // Картка: Програми тренувань
        Card(
            modifier = Modifier
                .width(250.dp)
                .height(100.dp)
                .clickable {
                    Log.d("LogTag", "Programs clicked")
                    context.startActivity(
                        Intent(context, FitnessProgramsActivity::class.java)
                    )
                },
            shape = RoundedCornerShape(100.dp),
            elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .background(MaterialTheme.colorScheme.surface)
                    .padding(16.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center
            ) {
                Image(
                    painter = painterResource(id = R.drawable.ic_exercise),
                    contentDescription = null,
                    modifier = Modifier.fillMaxWidth()
                )
                Text(
                    text = stringResource(id = R.string.edit_programs),
                    fontSize = 24.sp,
                    color = MaterialTheme.colorScheme.onSurface
                )
            }
        }

        // Картка: Редактор вправ
        Card(
            modifier = Modifier
                .width(250.dp)
                .height(100.dp)
                .clickable {
                    Log.d("LogTag", "New Exercises clicked")
                    context.startActivity(
                        Intent(context, ExerciseManagementActivity::class.java)
                    )
                },
            shape = RoundedCornerShape(100.dp),
            elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .background(MaterialTheme.colorScheme.surface)
                    .padding(16.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.Center
            ) {
                Image(
                    painter = painterResource(id = R.drawable.ic_exercise),
                    contentDescription = null,
                    modifier = Modifier.fillMaxWidth()
                )
                Text(
                    text = stringResource(id = R.string.exe_editor),
                    fontSize = 24.sp,
                    color = MaterialTheme.colorScheme.onSurface
                )
            }
        }
    }
}
